server/services/referenceData/referenceDataService.ts

```diff
@@ -1,31 +1,23 @@
-import logger from '../../logger'
-import { RestClientBuilder } from '../data'
-import ReferenceDataStore from '../data/referenceDataStore/referenceDataStore'
-import {
-  PersonIntegrationApiClient,
-  ProxyReferenceDataDomain,
-  ReferenceDataCodeDto,
-} from '../data/interfaces/personIntegrationApi/personIntegrationApiClient'
+import logger from '../../../logger'
+import ReferenceDataStore from '../../data/referenceDataStore/referenceDataStore'
+import { ReferenceDataCodeDto, ReferenceDataDomain } from '../../data/interfaces/referenceData'
+import { ReferenceDataSourceFactory } from './referenceDataSourceFactory'
 
 const REFERENCE_DATA_CACHE_TTL_HOURS = 1
 
-/**
- * Service class to retrieve and cache reference data from `hmpps-person-integration-api`.
- */
 export default class ReferenceDataService {
   constructor(
     private readonly referenceDataStore: ReferenceDataStore,
-    private readonly personIntegrationApiClientBuilder: RestClientBuilder<PersonIntegrationApiClient>,
+    private readonly referenceDataSourceFactory: ReferenceDataSourceFactory,
```

Comment:
The ReferenceDataService can now cater for all sources of reference data, and performs the caching for all sources now